File: python/problems/find_maximun_and_minimun/challenge.py
"""
Write a function to find the largest and smallest elements 
in an array.

input:
[1, 2, 3, 4, 5]

output:
(5, 1)
"""


def get_max_and_min(arr):
    min_value = float("inf")
    max_value = float("-inf")
    for num in arr:
        if num < min_value:
            min_value = num
        if num > max_value:
            max_value = num
    return (max_value, min_value)


def pythonic_get_max_and_min(arr):
    return (max(arr), min(arr))

File: javascript/problems/findMaximunAndMinimun/challenge.js
/**
Write a function to find the largest and smallest elements in an array.

input:
[1, 2, 3, 4, 5]

output:
(5, 1)
*/
function getLargestAndSmallest(arr) {
  let smallest = Number.MAX_SAFE_INTEGER
  let largest = Number.MIN_SAFE_INTEGER
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] > largest) {
      largest = arr[i];
    } if (arr[i] < smallest) {
      smallest = arr[i]
    }
  }
  return [largest, smallest]
}

module.exports = getLargestAndSmallest;
